src/schedule/zephyr_dma_domain.c

```diff
@@ -297,6 +297,16 @@ static int zephyr_dma_domain_register(struct ll_schedule_domain *domain,
 
 	tr_info(&ll_tr, "zephyr_dma_domain_register()");
 
+	/* don't even bother trying to register DMA IRQ for
+	 * non-registrable tasks.
+	 *
+	 * this is needed because zephyr_dma_domain_register() will
+	 * return -EINVAL for non-registrable tasks because of
+	 * register_dma_irq() which is not right.
+	 */
+	if (!pipe_task->registrable)
+		return 0;
```

Comment:
The function that calls domain_register() is zephyr_ll_task_schedule_common(). It makes no sense not to call domain_register() at all since timer domain doesn't skip such tasks. This is only a dma domain particularity. Also, calling domain_register() based on the type of domain from zephyr_ll_task_schedule_common() is much messier.